### ISSUE TITLE
[ONNX] test_op_consistency.py doesn't support constant inputs

### ISSUE THREAD

**Issue by titaiwangms** [pytorch/pytorch#102211]
### 🐛 Describe the bug

The current op test design in TorchScript exporter assumes all inputs/args are model inputs, which JIT tracer would trace them all even if they are not tensors.

```python
class SingleOpModel(torch.nn.Module):
    """Test model to wrap around a single op for export."""

    def __init__(self, op, kwargs):
        super().__init__()
        self.operator = op
        self.kwargs = kwargs

    def forward(self, *args):
        return self.operator(*args, **self.kwargs)
```

This results in issue in ops like `scatter_reduce` which has `dim` in arguments. Usually, it is used like:
```python
input.scatter_reduce(0, index, x, reduce='amax')
```

So the dim wouldb't be traced. However, if it is passed in as model input, it will be traced, and inside symbolic function. We will not get prim::Constant of dim.

cc @justinchuby 

### Versions

Nightly
